klarna-on-demand/src/main/java/com/klarna/ondemand/PreferencesActivity.java

```diff
@@ -0,0 +1,45 @@
+package com.klarna.ondemand;
+
+import com.jockeyjs.Jockey;
+import com.jockeyjs.JockeyImpl;
+
+import junit.framework.Assert;
+
+import java.util.Map;
+
+/**
+ * Responsible for showing and modifying a user's payment methods.
+ */
+public class PreferencesActivity extends WebViewActivity {
+
+    /**
+     * Token that uniquely identifies the user.
+     * This extra item is required.
+     */
+    public static final String EXTRA_USER_TOKEN = "userToken";
+
+    @Override
+    protected String getUrl() {
+        String token = getIntent().getStringExtra(EXTRA_USER_TOKEN);
+        if(token == null) {
+            throw new IllegalStateException("EXTRA_USER_TOKEN is not set for preferences activity");
+        }
+        return UrlHelper.preferencesUrl(token);
+    }
+
+    @Override
+    protected void handleUserReadyEventWithPayload(Map<Object, Object> payload) {
```

Comment:
I second that motion. Saying "WithPayload" when you have a payload argument is somewhat redundant.